src/include/cell.h

```diff
@@ -144,37 +144,65 @@ struct __wt_cell {
     uint8_t __chunk[1 + 1 + 1 + 7 * WT_INTPACK64_MAXSIZE + WT_INTPACK32_MAXSIZE];
 };
 
+/* AUTOMATIC FLAG VALUE GENERATION START */
+#define WT_CELL_UNPACK_OVERFLOW 0x1u            /* cell is an overflow */
+#define WT_CELL_UNPACK_PREPARE 0x2u             /* cell is part of a prepared transaction */
+#define WT_CELL_UNPACK_TIME_WINDOW_CLEARED 0x4u /* time window cleared because of restart */
+                                                /* AUTOMATIC FLAG VALUE GENERATION STOP */
+
 /*
- * WT_CELL_UNPACK --
- *	Unpacked cell.
+ * We have three flavors of cell unpack structures: one that can unpack any kind of cell, one that
+ * can only unpack address cells and one that can only unpack value cells. The reason is because we
+ * want to catch code using the wrong time window information relative to the cell type, minimize
+ * the size of the unpack structure, and finally reduce the code executed in performance paths,
+ * while still allowing other code such as verification and debug functions to be able to read any
```

Comment:
Does verification and debug functions know what type of cells they are reading?

Reply:
Yes -- you're correct, this was a comment left over from an earlier version of this code where the debug & verification code hadn't been fixed.